backend/cmm.ml

```diff
@@ -26,6 +26,11 @@ let typ_void = ([||] : machtype_component array)
 let typ_val = [|Val|]
 let typ_addr = [|Addr|]
 let typ_int = [|Int|]
+let typ_int32 = [|Int|]
+(* XXX ASZ: What layouts version does this go with? *)
+(* CR layouts XXX ASZ: Change [typ_int32] to fit in a 32-bit register, rather
+   than taking up a full 64-bit one. *)
+let typ_int64 = if Arch.size_int = 4 then [|Int;Int|] else [|Int|]
```

Comment:
Instead of having the complication of thinking about these multi-element arrays, I would just assert that Arch.size_int is 8.  It is very unlikely it will ever be 4 again.